configure

```diff
@@ -0,0 +1,363 @@
+  TEST_LDFLAGS_EXTRA            If you want to use some arguments in Linker for testing.
+                                  It's passed after TEST_LDFLAGS without overwrite it.
+
+Note: be careful about passing a path with space as prefix/eprefix/sbindir/mandir:
+for this thing You need to use backslash (\) with double quotation (") for every space.
+For example: ./configure --sbindir="/usr/bin\ with\ space/"
+
+See BUILD file for more information.
+EOF
+    exit 0
+}
+
+## Flag praser
+for arg in "$@"; do
+    case "$arg" in
+        --help) usage ;;
+        --quiet)
+            info() { true; }
+            sub_info() { true; }
+            warning() { true; }
+            sub_warning() { true; }
+        ;;
+        --clean) rm -f test* & rm -f mconfig && exit 0 ;;
+        --platform=*) PLATFORM="${arg#*=}" && shift ;;
+        --prefix=*) PREFIX="${arg#*=}" && shift ;;
+        --exec-prefix=*) EPREFIX="${arg#*=}" && shift;;
+        --sbindir=*) SBINDIR="${arg#*=}" && shift ;;
+        --mandir=*) MANDIR="${arg#*=}" && shift ;;
+        --syscontrolsocket=*) SYSCONTROLSOCKET="${arg#*=}" && shift ;;
+        --shutdown-prefix=*) SHUTDOWN_PREFIX="${arg#*=}" && shift ;;
+        --enable-shutdown|--enable-shutdown=yes) BUILD_SHUTDOWN=yes ;;
+        --disable-shutdown|--enable-shutdown=no) BUILD_SHUTDOWN=no ;;
+        --enable-cgroups|--enable-cgroups=yes) SUPPORT_CGROUPS=1 ;;
+        --disable-cgroups|--enable-cgroups=no) SUPPORT_CGROUPS=0 ;;
+        --enable-sanitize|--enable-sanitize=yes) NO_SANITIZE=0 ;;
+        --disable-sanitize|--enable-sanitize=no) NO_SANITIZE=1 ;;
+        --enable-utmpx|--enable-utmpx=yes) USE_UTMPX=1 ;;
+        --disable-utmpx|--enable-utmpx=no) USE_UTMPX=0 ;;
+        *=*)
+            envvar=$(expr "x$arg" : 'x\([^=]*\)=')
+            envvar_value=$(expr "X$arg" : '[^=]*=\(.*\)') || envvar_value="" # Fix empty var bug
+            eval "$envvar=\$envvar_value"
+            export "${envvar?}"
+        ;;
+        *) warning Unknown argument: "$arg" ;;
+    esac
+done
+
+## General Defines
+: "${PLATFORM:=$(uname)}"
+: "${PREFIX:="/usr"}"
+: "${EPREFIX:="/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SBINDIR:="/sbin"}"
+: "${MANDIR:="/share/man/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SHUTDOWN_PREFIX:=""}"
+: "${CXXFLAGS_EXTRA:=""}"
+: "${TEST_CXXFLAGS:=""}"
+: "${TEST_CXXFLAGS_EXTRA:=""}"
+: "${CPPFLAGS:=""}"
+: "${LDFLAGS_EXTRA:=""}"
+: "${TEST_LDFLAGS:=""}"
+: "${TEST_LDFLAGS_EXTRA:=""}"
+if [ "$PLATFORM" = "Linux" ]; then
+    : "${BUILD_SHUTDOWN:=yes}"
+    : "${SUPPORT_CGROUPS:=1}"
+    : "${SYSCONTROLSOCKET:="/run/dinitctl"}"
+else
+    : "${BUILD_SHUTDOWN:=no}"
+    : "${SUPPORT_CGROUPS:=0}"
+    : "${SYSCONTROLSOCKET:="/var/run/dinitctl"}"
+fi
+
+## Resolve final paths
+SBINDIR="$EPREFIX/$SBINDIR"
+MANDIR="$PREFIX/$MANDIR"
+
+## Finalize $CXXFLAGS & $LDFLAGS
+if [ -z "${CXXFLAGS+IS_SET}" ]; then
+    CXXFLAGS=""
+    NO_CXXFLAGS="false"
```

Comment:
NO_CXXFLAGS gets set `false` when CXXFLAGS is unset - shouldn't it be `true` in that case? (or rename it `CXXFLAGS_SET` for example).

Reply:
Seems like `NO_CXXFLAGS` is not a good name. I rename it to `AUTO_CXXFLAGS`.